src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -74,7 +72,11 @@ internal sealed class TracerProviderSdk : TracerProvider
             }
         }
 
-        foreach (var processor in state.Processors)
+        var processors = (IEnumerable<BaseProcessor<Activity>>)state.Processors.OrderBy(p => (int)p.PipelineWeight);
```

Comment:
Is this a stable sorting algorithm or not?

Reply:
If we believe the [docs](https://learn.microsoft.com/dotnet/api/system.linq.enumerable.orderby), it is stable:

> This method performs a stable sort; that is, if the keys of two elements are equal, the order of the elements is preserved. In contrast, an unstable sort does not preserve the order of elements that have the same key.

Reply:
> If we believe the docs

How should I interpret it? - "YES" or "I'm not sure"

I think @cijothomas asked the same question https://github.com/open-telemetry/opentelemetry-dotnet/pull/5424/files#r1517025574. The answer we expect is either "YES, it is stable and it will keep the existing behavior" or "NO, I realized there is something wrong and I'll update the PR to fix it".

Reply:
It was really a question for you. Do you trust the docs? If yes, we're good. If no, I need to go and do more work 🤣

There's seems to be a test in place:

https://github.com/dotnet/runtime/blob/40ac297a1101ac293a1bae01f2e2e1556f7ec017/src/libraries/System.Linq/tests/OrderByTests.cs#L149-L172

But I also made our tests verify it: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5424/commits/62a9f40cdb55debe9a2aa6e42fe6c1f50e3faf67